components/Admin/Admin.js

```diff
@@ -20,7 +25,7 @@ class Admin extends React.Component {
   componentWillMount() {
     fetchLocations()
       .then(locations => {
-        this.setState({ locations })
+        this.setState({ loading: false, locations })
```

Comment:
I like reading "loading" but also, it's easy to forget to update when you finish fetching. I *hate* null, but it's the closest thing we have to `Maybe` or `Option`. So, what about locations starts and null and you provide a loading UI when it's null, and then when there's a list, you know it's been loaded?